system/scripts/zone/ais/sadhu_dummy.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Melia.Shared.Game.Const;
+using Melia.Zone;
+using Melia.Zone.Network;
+using Melia.Zone.Scripting;
+using Melia.Zone.Scripting.AI;
+using Melia.Zone.Skills;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Actors.Characters.Components;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using Yggdrasil.Logging;
+
+[Ai("SadhuDummy")]
+public class SadhuDummyAiScript : AiScript
+{
+	private const int MaxChaseDistance = 50;
+	private const int MaxMasterDistance = 80;
+
+	ICombatEntity target;
+
+	protected override void Setup()
+	{
+		During("Idle", CheckEnemies);
+		During("Attack", CheckTarget);
+		During("Attack", CheckMaster);
+	}
+
+	protected override void Root()
+	{
+		StartRoutine("Idle", Idle());
+	}
+
+	protected IEnumerable Idle()
+	{
+		var movement = this.Entity.Components.Get<MovementComponent>();
+
+		movement.SetMoveSpeedType(MoveSpeedType.Run);
+		
+		if (this.Entity is Character entityCharacter && entityCharacter.IsDummy)
+		{
+			SetRunning(true);
+			entityCharacter.Properties.Modify(PropertyName.MSPD_BM, 55);
```

Comment:
This hack is not ideal. `SetRunning` technically does something similar to what you wanted to do, but it's limited to mobs because characters don't differentiate between walking and running. Seeing that and copying the code into the AI script just causes unnecessary redundancies though.

It's also not clear what `SetRunning` and this are supposed to do together. One might think the former is still there in case this AI is used for a monster, but then it wouldn't belong in this if. And even if it did, you're also calling `SetMoveSpeedType`, which is the very method `SetRunning` is calling as well.

What I believe you probably want to do here, is call `SetFixedMoveSpeed`. That works for both characters and monsters, saves you from doing any of this manually, and you can simply set the entity's speed to that of the owner, plus maybe a little on top.